### PR TITLE
Storage Pods modelling and config generation

### PR DESCRIPTION
**Pull request by swagner-de**

### CODE REVIEW

File: networking_ccloud/tools/netbox_config_gen.py [hunk 1]
```diff
@@ -243,16 +259,79 @@ def get_svi_ips_per_vlan(self, device: NbRecord) -> Dict[NbRecord, List[NbRecord
...
+        if vlan.group.slug.startswith('cc-vpod'):  # type: ignore
+            # we would actually like to use the vlan group's scope type,
+            # yet netbox only allows a per rack/site/region scope type
+            # but out vpods span over multiple racks, so we have to use
+            # a site scope type and need to fallback on regex parsing here
+
+            # expects something like cc-vpod271
+            m = re.match(r'.*?(\d+)$', vlan.group.slug)  # type: ignore
+            if not m:
+                raise ConfigException(f'vlan group {vlan.group.id} must end with digits '  # type: ignore
```
Comment (sebageek) [inline networking_ccloud/tools/netbox_config_gen.py]: Couldn't you also print out the `vlan.group.slug` here to show what you tried to match?

File: networking_ccloud/tools/netbox_config_gen.py [hunk 2]
```diff
@@ -243,16 +259,79 @@ def get_svi_ips_per_vlan(self, device: NbRecord) -> Dict[NbRecord, List[NbRecord
...
         infra_nets = set()
+        extra_vlans = set()
         if iface.lag:
             iface = iface.lag
 
         # FIXME: support untagged VLANs
         # FIXME: support DHCP relay
         for vlan in getattr(iface, 'tagged_vlans', list()):
+            # some infra vlans we will not manage, just put in the allowed VLAN list
+            tags = set((x.slug for x in vlan.tags))
```
Comment (sebageek) [inline networking_ccloud/tools/netbox_config_gen.py]: No need to doublebag

File: networking_ccloud/tools/netbox_config_gen.py [hunk 3]
```diff
@@ -562,17 +645,26 @@ def purge_infranetworks_on_metagroup(cls, metagroup: conf.Hostgroup, member_hgs:
             raise ValueError("'member_hgs.binding_host[0]' and metagroup.members are not matching")
 
         minimum_common_infra_nets = set(member_hgs[0].infra_networks)
+        minimum_common_extra_vlans = set(member_hgs[0].extra_vlans if member_hgs[0].extra_vlans else [])
```
Comment (sebageek) [inline networking_ccloud/tools/netbox_config_gen.py]: could be `minimum_common_extra_vlans = set(member_hgs[0].extra_vlans or []`, if you want it shorter.

File: networking_ccloud/tools/netbox_config_gen.py [hunk 4]
```diff
@@ -562,17 +645,26 @@ def purge_infranetworks_on_metagroup(cls, metagroup: conf.Hostgroup, member_hgs:
...
+            minimum_common_extra_vlans.intersection_update(member.extra_vlans if member.extra_vlans else [])
 
         # purge them from the member_hgs
         for member in member_hgs:
             remaining_infra_nets = list()
+            remaining_extra_vlans = list()
             for infra_net in member.infra_networks:
                 if infra_net not in minimum_common_infra_nets:
                     remaining_infra_nets.append(infra_net)
+            for extra_vlan in member.extra_vlans if member.extra_vlans else []:
```
Comment (sebageek) [inline networking_ccloud/tools/netbox_config_gen.py]: Could also be `for extra_vlan in member.extra_vlans or []` if you want.

File: networking_ccloud/tools/netbox_config_gen.py (hunk 3 above)
Comment (swagner-de) [reply networking_ccloud/tools/netbox_config_gen.py]: I did not know. Thanks.

File: networking_ccloud/tools/netbox_config_gen.py [hunk 5]
```diff
@@ -243,16 +257,79 @@ def get_svi_ips_per_vlan(self, device: NbRecord) -> Dict[NbRecord, List[NbRecord
...
         infra_nets = set()
+        extra_vlans = set()
         if iface.lag:
             iface = iface.lag
 
         # FIXME: support untagged VLANs
         # FIXME: support DHCP relay
         for vlan in getattr(iface, 'tagged_vlans', list()):
+            # some infra vlans we will not manage, just put in the allowed VLAN list
+            tags = set((x.slug for x in vlan.tags))
```
Comment (sebageek) [inline networking_ccloud/tools/netbox_config_gen.py]: Was that something you still wanted to change or should I stop mentioning it? (double brackets)

**Review by sebageek — APPROVED** [review]

**Pull request merged**